File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        s = s.strip().split()
        # s is no longer a string, but a list of strings. Each word is an element in this list.
        
        i = 0
        j = len(s) - 1
        
        while i < j:
            s[i], s[j] = s[j], s[i]
            i += 1
            j -= 1
        
        return " ".join(s)

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String[] words = s.trim().split("\\s+");
        
        int i = 0;
        int j = words.length - 1;
        while(i < j){
            String tmp = words[i];
            words[i] = words[j];
            words[j] = tmp;
            i++;
            j--;
        }
        
        return String.join(" ", words);  
    }
}